packages/vx-axis/src/types.ts

```diff
@@ -49,75 +54,59 @@ export type SharedAxisProps<ScaleInput> = {
   /** The class name applied to each tick group. */
   tickClassName?: string;
+  /** Override the component used to render tick labels (instead of <Text /> from @vx/text) */
+  tickComponent?: (tickRendererProps: TickRendererProps) => React.ReactNode;
   /** A [d3 formatter](https://github.com/d3/d3-scale/blob/master/README.md#continuous_tickFormat) for the tick text. */
-  tickFormat?: TickFormatter<ScaleInput>;
+  tickFormat: TickFormatter<ScaleInput<Scale>>;
   /** A function that returns props for a given tick label. */
-  tickLabelProps?: TickLabelProps<ScaleInput>;
+  tickLabelProps?: TickLabelProps<ScaleInput<Scale>>;
   /** The length of the tick lines. */
-  tickLength?: number;
+  tickLength: number;
   /** The color for the tick's stroke value. */
   tickStroke?: string;
   /** A custom SVG transform value to be applied to each tick group. */
   tickTransform?: string;
-  /** An array of values that determine the number and values of the ticks. Falls back to `scale.ticks()` or `.domain()`. */
-  tickValues?: ScaleInput[];
-  /** Override the component used to render tick labels (instead of <Text /> from @vx/text) */
-  tickComponent?: (tickRendererProps: TickRendererProps) => React.ReactNode;
-  /** A top pixel offset applied to the entire axis. */
-  top?: number;
-  /** For more control over rendering or to add event handlers to datum, pass a function as children. */
-  children?: (renderProps: ChildRenderProps<ScaleInput>) => React.ReactNode;
-};
-
-// In order to plot values on an axis, Output must be numeric or coercible to a number.
-// Some scales return undefined.
-export type ScaleOutput = number | { valueOf(): number } | undefined;
-
-export type GenericScale<ScaleInput> =
-  | ScaleNoRangeRound<ScaleInput>
-  | ScaleWithRangeRound<ScaleInput>;
-
-interface ScaleNoRangeRound<ScaleInput> {
-  (value: ScaleInput): ScaleOutput | [ScaleOutput, ScaleOutput]; // quantize scales return an array
-  domain(): ScaleInput[] | [ScaleInput, ScaleInput];
-  domain(scaleInput: ScaleInput[] | [ScaleInput, ScaleInput]): any; // we can't capture the copy of the type accurately
-  range(): ScaleOutput[] | [ScaleOutput, ScaleOutput];
-  range(scaleOutput: ScaleOutput[] | [ScaleOutput, ScaleOutput]): any;
-  ticks?: (count: number) => ScaleInput[] | [ScaleInput, ScaleInput];
-  bandwidth?: () => number;
-  round?: () => boolean;
-  tickFormat?: () => (input: ScaleInput) => FormattedValue;
-  copy(): this;
 }
 
-// We cannot have optional methods AND overloads, so define a separate type for rangeRound
-interface ScaleWithRangeRound<ScaleInput> extends ScaleNoRangeRound<ScaleInput> {
-  rangeRound(): ScaleOutput[] | [ScaleOutput, ScaleOutput];
-  rangeRound(scaleOutput: ScaleOutput[] | [ScaleOutput, ScaleOutput]): any;
-}
-
-export interface Point {
+interface Point {
   x: number;
   y: number;
 }
 
-export type ChildRenderProps<ScaleInput> = {
+export type AxisRendererProps<Scale extends AxisScale> = CommonProps<Scale> & {
```

Comment:
Instead of accepting `ScaleInput` generic, accept `Scale` with the constraint that it must be a scale compatible for axis (compatible meaning scale output is `AxisScaleOutput = number | NumberLike | undefined` regardless of input type) then infer `ScaleInput` from `Scale`